packages/rum-core/test/performance-monitoring/performance-monitoring.spec.js

```diff
@@ -678,4 +678,46 @@ describe('PerformanceMonitoring', function() {
       jasmine.any(Function)
     ])
   })
+
+  it('should create http-request transaction if no current transaction exist', done => {
+    const transactionService = serviceFactory.getService('TransactionService')
+
+    var req = new window.XMLHttpRequest()
+    req.open('GET', '/')
+
+    var task = {
+      source: XMLHTTPREQUEST,
+      data: {
+        target: req,
+        method: 'GET'
+      }
+    }
+
+    spyOn(transactionService, 'startTransaction').and.callThrough()
+
+    performanceMonitoring.processAPICalls('schedule', task)
+    expect(transactionService.startTransaction).toHaveBeenCalledWith(
+      'GET /',
+      'http-request',
+      jasmine.objectContaining({ managed: true })
+    )
+
+    var tr = transactionService.getCurrentTransaction()
+    expect(tr).toBeDefined()
+
+    setTimeout(() => {
```

Comment:
Instead of depending on a timeout, cleaner approach would be to use the `transaction:end` event to do the check for the payload errors.

Reply:
@vigneshshanmugam , the setTimeout here is to simulate the latency of a XHR. It's not for the transaction to end.

Reply:
Ahh, Got you. cant we do req.send and listen for the `transaction:end` event instead of timeouts. 

Reason why i am so against timeouts is that, its not reliable and would hang the test even if there is a bug in a depedency.

Reply:
IMO, it's fine to use setTimeout as long as we properly call the `done` function. Not sure why just using setTimeout has to make the tests unreliable.

Reply:
Just fake one of the timers somewhere(can happen in dependencies or in our code) and we will never see the tests calling done. we can leave it to `setTimeout`,not a huge deal.